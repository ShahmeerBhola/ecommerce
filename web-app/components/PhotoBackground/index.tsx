import React from 'react';

type Props = {
  image?: string;
  className?: string;
  cover?: boolean;
  styles?: React.CSSProperties;
};

const PhotoBackground: React.FunctionComponent<Props> = ({ children, image, cover = true, className = '', styles }) => {
  let backgroundStyles: React.CSSProperties = {};

  if (image) {
    backgroundStyles.backgroundImage = `url(${image})`;
  }
  if (styles !== undefined) {
    backgroundStyles = { ...backgroundStyles, ...styles };
  }

  return (
    <div className={`bg-no-repeat ${className} ${cover ? 'bg-cover' : ''}`} style={backgroundStyles}>
      {children}
    </div>
  );
};

export default PhotoBackground;
