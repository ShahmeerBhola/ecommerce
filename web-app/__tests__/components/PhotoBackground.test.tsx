import React from 'react';
import renderer from 'react-test-renderer';
import PhotoBackground from '../../components/PhotoBackground';

describe('PhotoBackground component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <PhotoBackground image="my-truck.jpg">
          <p>Hello</p>
        </PhotoBackground>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with className props', () => {
    const tree = renderer
      .create(
        <PhotoBackground image="my-truck2.jpg" className="hello-there">
          <p>Hello</p>
        </PhotoBackground>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when cover prop is false', () => {
    const tree = renderer
      .create(
        <PhotoBackground image="my-truck3.jpg" cover={false}>
          <p>Hello</p>
        </PhotoBackground>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when cover prop is true', () => {
    const tree = renderer
      .create(
        <PhotoBackground image="my-truck3.jpg" cover>
          <p>Hello</p>
        </PhotoBackground>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
