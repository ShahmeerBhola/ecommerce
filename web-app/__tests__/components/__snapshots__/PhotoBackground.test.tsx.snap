// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoBackground component it renders properly 1`] = `
<div
  className="bg-no-repeat bg-cover"
  style={
    Object {
      "backgroundImage": "url(my-truck.jpg)",
    }
  }
>
  <p>
    Hello
  </p>
</div>
`;

exports[`PhotoBackground component it renders properly when cover prop is false 1`] = `
<div
  className="bg-no-repeat"
  style={
    Object {
      "backgroundImage": "url(my-truck3.jpg)",
    }
  }
>
  <p>
    Hello
  </p>
</div>
`;

exports[`PhotoBackground component it renders properly when cover prop is true 1`] = `
<div
  className="bg-no-repeat bg-cover"
  style={
    Object {
      "backgroundImage": "url(my-truck3.jpg)",
    }
  }
>
  <p>
    Hello
  </p>
</div>
`;

exports[`PhotoBackground component it renders properly with className props 1`] = `
<div
  className="bg-no-repeat hello-there bg-cover"
  style={
    Object {
      "backgroundImage": "url(my-truck2.jpg)",
    }
  }
>
  <p>
    Hello
  </p>
</div>
`;
